# Question 1 results:

## Calculate the following for all sequences ≤ 100kb and all sequences > 100kb:
fileformat: 100k-prefix is for all sequences bigger than 100k and s100k-prefix is for all sequences smaller or equal to 100k:
  Total number of nucleotides : 100ktotalnucleotidecount.txt and s100ktotalnucleotidecount.txt
  
  Total number of Ns: 100kNcount.txt and s100kNcount.txt
  
  Total number of sequences : 100ksequencecount.txt and s100ksequencecount.txt

## Plots of the following for the whole genome, for all sequences ≤ 100kb, and all sequences > 100kb:
fileformat: 100k-prefix is for all sequences bigger than 100k and s100k-prefix is for all sequences smaller or equal to 100k:

  Sequence length distribution: 100klength.pdf and s100klength.pdf
  
  Sequence GC% distribution: 100kgc.pdf and s100kgc.pdf
  
  Cumulative genome size sorted from largest to smallest sequences: 100kgs.pdf and s100kgs.pdf

# Assembly assessment

## comparison of n50 : n50comparison.txt

## Compare your assembly to the contig assembly (not the scaffold assembly!) from Drosophila melanogaster on FlyBase using a dotplot constructed with MUMmer (Hint: use faSplitByN as demonstrated in class) : image.png

## Compare your assembly to both the contig assembly and the scaffold assembly from the Drosophila melanogaster on FlyBase using a contiguity plot (Hint: use plotCDF2 as demonstrated in class and see this example): r6_v_truseq.png


## Calculate BUSCO scores of both assemblies and compare them : full_table_unitigs_diptera_odb9.tsv
